eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/ConnectionParameters.java

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.eventhubs;
+
+import com.azure.core.amqp.TransportType;
+import com.azure.eventhubs.implementation.TokenProvider;
+import reactor.core.scheduler.Scheduler;
+
+import java.time.Duration;
+import java.util.Objects;
+
+/**
+ * A wrap class that contains all parameters that need to establish a connection to an event hub.
+ */
+class ConnectionParameters {
+    private final CredentialInfo credentials;
+    private final Duration timeout;
+    private final TokenProvider tokenProvider;
+    private final TransportType transport;
+    private final Retry retryPolicy;
+    private final ProxyConfiguration proxyConfiguration;
+    private final Scheduler scheduler;
+
+    ConnectionParameters(final CredentialInfo credentials, final Duration timeout, final TokenProvider tokenProvider,
+                         final TransportType transport, final Retry retryPolicy,
+                         final ProxyConfiguration proxyConfiguration, final Scheduler scheduler) {
+        Objects.requireNonNull(credentials);
+        Objects.requireNonNull(timeout);
+        Objects.requireNonNull(tokenProvider);
+        Objects.requireNonNull(transport);
+        Objects.requireNonNull(retryPolicy);
+        Objects.requireNonNull(proxyConfiguration);
+        Objects.requireNonNull(scheduler);
+
+        this.credentials = credentials;
+        this.timeout = timeout;
+        this.tokenProvider = tokenProvider;
+        this.transport = transport;
+        this.retryPolicy = retryPolicy;
+        this.proxyConfiguration = proxyConfiguration;
+        this.scheduler = scheduler;
+    }
+
+    CredentialInfo getCredentials() {
```

Comment:
Where's `eventHubPath()`? Also, this is a model class, we don't need the `get`.

Reply:
eventHubPath? Do you mean Entity_path? if you refer to Entity_Path, it is a property field in CredentialInfo.

Reply:
EventHubPath ~= EntityPath. If you have a connection string from the portal that links to a specific event hub, it will contain EntityPath. However, if you get the connection string from the Event Hubs namespace (such as the default one "RootManageSharedAccessKey"), it won't include EntityPath. In this case, we should expose EventHubPath in the builder, so users can set it and choose a specific event hub to connect to.